src/deepsparse/pipeline.py

```diff
@@ -1353,3 +1361,76 @@ def zero_shot_text_classification_pipeline(*args, **kwargs) -> "Pipeline":
         with 2 streams to make use of parallel inference of labels
     """
     return Pipeline.create("zero_shot_text_classification", *args, **kwargs)
+
+
+def _dynamic_import_task(module: str) -> str:
+    """
+    Dynamically imports `module` with importlib, and returns the `TASK`
+    attribute on the module (something like `importlib.import_module(module).TASK`).
+
+    Example contents of `module`:
+    ```python
+    from deepsparse.pipeline import Pipeline
+    from deepsparse.transformers.pipelines.question_answering import (
+        QuestionAnsweringPipeline,
+    )
+
+    TASK = "my_qa_task"
+    Pipeline.register(TASK)(QuestionAnsweringPipeline)
+    ```
+
+    NOTE: this modifies `sys.path`.
+
+    :raises FileNotFoundError: if path does not exist
+    :raises RuntimeError: if the imported module does not contain `TASK`
+    :raises RuntimeError: if the module doesn't register the task
+    :return: The task from the imported module.
+    """
+    parent_dir, module_name = _module_to_dir_and_name(module)
```

Comment:
what we can do to allow for more flexible locations/ways to provide the custom module file.  Right now it seems like the file needs to be nested somewhere inside the directory the python process is called from? 

Thoughts on trying to import first (ie user would have already exported the custom file into the python path) and then trying to dynamically find and add the file to path on fail?

Reply:
It should allow any path on the file system - that's what the `sys.path.append` will do. E.g. if you specify `import:/home/corey/deepsparse/a.py` parent_dir will be `/home/corey/deepsparse` and it will should add it to the python path

Reply:
err i forgot we changed to module names instead of paths, so the above should be `import:home.corey.deepsparse.a`

Reply:
ah yeah makes sense - still though, it looks like that would translate to `home/corey/deepsparse` and `a.py` (rather than having the preceding `/` to home)

Reply:
yeah great call out - missing that test case. in this case they'd have to add a '.' in front of path i think, which is a bit awkward. thoughts on going back to file paths? I think that makes it a bit more clear what a user can enter

Reply:
let detect and support either two cases:

1. user gives `module` path (ie `.` separated, does not end in `.py`) -> we assume the module is valid in the pythonpath and run importlib
2. user gives file path (ends in `.py`, optionally '/' separated) -> we append the file to python path and then run importlib

Reply:
I like it, will do!

Reply:
Right now it only keys off of `endswith(".py")` - depending on anything else seemed to have too many edge cases

Reply:
looks good